tiledb/sm/dimension_label/dimension_label.cc

```diff
@@ -118,37 +120,77 @@ void DimensionLabel::open(
     EncryptionType encryption_type,
     const void* encryption_key,
     uint32_t key_length) {
+  // Open indexed array and get schema
   throw_if_not_ok(indexed_array_->open(
       query_type,
       timestamp_start,
       timestamp_end,
       encryption_type,
       encryption_key,
       key_length));
+  auto&& [index_status, indexed_array_schema] =
+      indexed_array_->get_array_schema();
+  throw_if_not_ok(index_status);
+  // Open labelled array and get schema
   throw_if_not_ok(labelled_array_->open(
       query_type,
       timestamp_start,
       timestamp_end,
       encryption_type,
       encryption_key,
       key_length));
-  load_schema();
+  auto&& [label_status, labelled_array_schema] =
+      labelled_array_->get_array_schema();
+  throw_if_not_ok(label_status);
+  load_schema(indexed_array_schema.value(), labelled_array_schema.value());
   query_type_ = query_type;
+  // Restrict reading to only shared fragments.
+  if (query_type_ == QueryType::READ) {
+    if (is_remote_) {
```

Comment:
- I'm not sure why this wouldn't work
- When we discussed "remote" dimension labels it had a different connotation ("external to the dimension label directory"). Since this is new code, I'd like to see a more descriptive variable like `is_tiledb_rest`, but ideally we should try to avoid backend-specific code here, if at all possible.

Reply:
I agree we should not have backend specific code. I'd also like to understand why we can't support rest here? REST is a hard product requirement so I want to understand the gap and the plan to get this addressed.

Reply:
I am using `Array::array_directory()` to get the fragment URIs. As far as I could tell by inspecting `array.cc`, the `array_dir_` private member that `Array::array_directory()` returns is never fully initialized for arrays that are accessed via REST.

Assuming I'm not mistaken, in the short term I can throw an error inside `array_directory` if the object is not fully initialized, then re-throw with a more specific error in `DimensionLabel`. Or I can just update the `is_remote_` private variable to `is_tiledb_rest_`. Preference?